apps/avifenc.c

```diff
@@ -1634,6 +1848,144 @@ int main(int argc, char * argv[])
     }
 #endif
 
+    if (memcmp(&pendingSettings, &emptySettingsReference, sizeof(avifInputFileSettings)) != 0) {
+        fprintf(stderr, "WARNING: Trailing options with update suffix has no effect. Place them before the input you intend to apply to.\n");
+    }
+
+    if (settings.progressive && input.filesCount > 1) {
+        fprintf(stderr, "ERROR: --progressive only supports one input.\n");
+        returnCode = 1;
+        goto cleanup;
+    }
+
+    for (int i = 0; i < input.filesCount; ++i) {
+        avifInputFile * file = &input.files[i];
+        avifInputFileSettings * fileSettings = &file->settings;
+
+        if (fileSettings->autoTiling.set) {
+            if (fileSettings->tileRowsLog2.set || fileSettings->tileColsLog2.set) {
+                fprintf(stderr, "ERROR: --autotiling is specified but --tilerowslog2 or --tilecolslog2 is also specified for current input\n");
+                returnCode = 1;
+                goto cleanup;
+            }
+            // At here this entry can only be set by command line, so its value have to be AVIF_TRUE.
```

Comment:
I had a hard time understanding this sentence, maybe make it extra verbose?

```c
assert(fileSettings->autoTiling.value);  // There is no --noautotiling flag.
// Disable manual tiling (possibly set by previous input).
```

Reply:
Put checks on tiling paramters together (should be from the beginning but was somewhat missed), and added some comments. Hope this explains it better.

In general this (switching between auto and manual tiling during encoding) should be a very rare case.